commands/command_track.go

```diff
@@ -74,32 +78,43 @@ ArgsLoop:
 			}
 		}
 
-		encodedArg := strings.Replace(pattern, " ", "[[:space:]]", -1)
-		_, err := attributesFile.WriteString(fmt.Sprintf("%s filter=lfs diff=lfs merge=lfs -text\n", encodedArg))
-		if err != nil {
-			Print("Error adding path %s", pattern)
-			continue
+		if !trackDryRunFlag {
+			encodedArg := strings.Replace(pattern, " ", "[[:space:]]", -1)
+			_, err := attributesFile.WriteString(fmt.Sprintf("%s filter=lfs diff=lfs merge=lfs -text\n", encodedArg))
+			if err != nil {
+				Print("Error adding path %s", pattern)
+				continue
+			}
 		}
 		Print("Tracking %s", pattern)
 
-		// Make sure any existing git tracked files have their timestamp updated
-		// so they will now show as modifed
-		// note this is relative to current dir which is how we write .gitattributes
-		// deliberately not done in parallel as a chan because we'll be marking modified
-		gittracked, err := git.GetTrackedFiles(pattern)
-		if err != nil {
-			LoggedError(err, "Error getting git tracked files")
-			continue
-		}
-		now := time.Now()
-		for _, f := range gittracked {
-			err := os.Chtimes(f, now, now)
+		if !trackNoTouchFlag || trackDryRunFlag {
```

Comment:
I actually question whether the `--no-touch` option is a good idea at all in fact. `track` only touches files when it absolutely has to, and *not* touching them in this case results in really nasty problems down the line, because `.gitattributes` wants to run the filters but the content of the files in the git commit doesn't reflect that. The only case where you'd be avoiding touching files with this option will probably cause more problems than it solves.